openhands/runtime/impl/eventstream/eventstream_runtime.py

```diff
@@ -106,6 +109,11 @@ def __init__(
         attach_to_existing: bool = False,
         headless_mode: bool = True,
     ):
+        global _atexit_registered
+        if not _atexit_registered:
+            _atexit_registered = True
+            atexit.register(remove_all_runtime_containers)
```

Comment:
ahh thank you for fixing this

Reply:
I took it out to a separate PR for ease of reviewing :)

Reply:
https://github.com/All-Hands-AI/OpenHands/pull/5712